src/Cli/dotnet/commands/dotnet-workload/install/FileBasedInstaller.cs

```diff
@@ -376,37 +398,17 @@ public void GarbageCollect(Func<string, IWorkloadResolver> getResolverForWorkloa
                     }
                 }
 
-                string installationRecordPath = null;
                 foreach (var featureBandToRemove in featureBandsToRemove)
                 {
-                    installationRecordPath = GetManifestInstallRecordPath(manifestId, manifestVersion, manifestFeatureBand, featureBandToRemove);
-                    File.Delete(installationRecordPath);
+                    RemoveManifestInstallationRecord(manifestId, manifestVersion, manifestFeatureBand, featureBandToRemove);
                 }
 
-                if (installationRecordPath != null)
+                if (featureBandsToRemove.Count == manifestInstallRecords[(manifestId, manifestVersion, manifestFeatureBand)].Count)
                 {
-                    var installationRecordDirectory = Path.GetDirectoryName(installationRecordPath);
-                    if (!Directory.GetFileSystemEntries(installationRecordDirectory).Any())
-                    {
-                        //  There are no installation records for the workload manifest anymore, so we can delete the manifest
-                        _reporter.WriteLine(string.Format(LocalizableStrings.DeletingWorkloadManifest, manifestId, $"{manifestVersion}/{manifestFeatureBand}"));
-                        var manifestPath = Path.Combine(GetManifestInstallDirForFeatureBand(manifestFeatureBand.ToString()), manifestId.ToString(), manifestVersion.ToString());
-                        Directory.Delete(manifestPath, true);
-
-                        //  Delete empty manifest installation record directory, and walk up tree deleting empty directories to clean up
-                        Directory.Delete(installationRecordDirectory);
-
-                        var manifestVersionDirectory = Path.GetDirectoryName(installationRecordDirectory);
-                        if (!Directory.GetFileSystemEntries(manifestVersionDirectory).Any())
-                        {
-                            Directory.Delete(manifestVersionDirectory);
-                            var manifestIdDirectory = Path.GetDirectoryName(manifestVersionDirectory);
-                            if (!Directory.GetFileSystemEntries(manifestIdDirectory).Any())
-                            {
-                                Directory.Delete(manifestIdDirectory);
-                            }
-                        }
-                    }
+                    //  All installation records for the manifest were removed, so we can delete the manifest
+                    _reporter.WriteLine(string.Format(LocalizableStrings.DeletingWorkloadManifest, manifestId, $"{manifestVersion}/{manifestFeatureBand}"));
+                    var manifestPath = Path.Combine(GetManifestInstallDirForFeatureBand(manifestFeatureBand.ToString()), manifestId.ToString(), manifestVersion.ToString());
```

Comment:
Why don't we have to delete the install records anymore? Wouldn't leaving old ones behind mean we wouldn't be able to delete it if it's reinstalled later?